### DISCUSSION TITLE
Is there a way to speed up page load in South America and Eastern Europe?

### DISCUSSION THREAD

**Discussion by zachguo** [vercel/community#453] [Help] [answered]
The page-load time of my Vercel deployment(mostly statically generated) is consistently slower by 80%~100% in South America, Eastern Europe and some Southeast Asian nations.
Is this due to the lack of edge CDN nodes in those regions? or is there anything I can do?

**Comment by leerob** [comment, marked as answer]
Are you server rendering your site or using static generation? What framework or tool?

**Reply by zachguo** [reply]
React.js, mostly SSG, and some SSR for search pages. Load speed is slower for both in those regions.

I drill down further into my performance data, and found slowness is mainly caused by `Request` and `Processing` stages. In SE Asian nations, median `Request` latency is usually decent (around 100ms) but median `Processing` takes 1X longer in comparison to US and JP. I guess it's because users have less powerful devices for rendering pages. However, for some SA and EE countries, `Request` latency can reach 200-500 ms, plus longer `Processing` time.

**Reply by leerob** [reply]
Are you using Next.js or some other SSR React solution? What kind of data are you fetching, and where it is located at?

**Reply by zachguo** [reply]
Basically vanilla next.js and vercel without much custom configuration except a few redirects for old links. Zero DB connection or client-side fetch. For SSG & SSR, data is from a few static json files.

**Reply by leerob** [reply]
Got it. If you're not fetching from anywhere, you probably don't need to do SSR and can make everything static, which should help improve performance 😄

**Reply by zachguo** [reply]
Yup, I use SSG whenever I can because SSR is slow even for North America or East Asia.
However, SSG is 1X slower in SA, E EU, SE Asia.

**Reply by Pieparker** [reply]
Certainly could be attributable to throughput/distance from a CDN node. We'd love to see this as a [feature suggestion](https://github.com/vercel/community/discussions/new?category=feature-suggestions)!
